packages/realm/src/Realm.ts

```diff
@@ -786,13 +787,17 @@ export class Realm {
     values: Partial<T> | Partial<RealmInsertionModel<T>>,
     mode: UpdateMode.All | UpdateMode.Modified | boolean,
   ): RealmObject<T> & T;
-  create<T extends RealmObject>(type: Constructor<T>, values: RealmInsertionModel<T>, mode?: UpdateMode.Never): T;
-  create<T extends RealmObject>(
+  create<T extends AnyRealmObject = RealmObject & DefaultObject>(
+    type: Constructor<T>,
+    values: RealmInsertionModel<T>,
+    mode?: UpdateMode.Never,
+  ): T;
+  create<T extends AnyRealmObject = RealmObject & DefaultObject>(
```

Comment:
Is the default really needed here? I imagine it'll be better to leave it out and rely on type inference from the first argument.